src/measure_theory/function/lp_space.lean

```diff
@@ -1234,28 +1252,50 @@ begin
   exact ennreal.coe_ne_top,
 end
 
-lemma snorm_ess_sup_const_smul {f : α → F} (c : 𝕜) :
-  snorm_ess_sup (c • f) μ = (‖c‖₊ : ℝ≥0∞) * snorm_ess_sup f μ :=
-by simp_rw [snorm_ess_sup,  pi.smul_apply, nnnorm_smul, ennreal.coe_mul, ennreal.ess_sup_const_mul]
+lemma ess_sup_comp_le {g : F → E} {K : ℝ≥0}
+  (hg : lipschitz_with K g) (hg0 : g 0 = 0) (f : α → F) :
+  snorm_ess_sup (g ∘ f) μ ≤ K • snorm_ess_sup f μ :=
+calc ess_sup (λ x, ↑‖g (f x)‖₊) μ ≤ ess_sup (λ x, ↑(K • ‖f x‖₊)) μ :
+    ess_sup_mono_ae (ae_of_all _ $ λ a, ennreal.coe_le_coe.mpr $
+      -- TODO: add `lipschitz_with.nnnorm_sub_le` and `lipschitz_with.nnnorm_le`
+      by simpa [hg0] using hg.norm_sub_le (f a) 0)
+  ... = _ : by simp_rw [ennreal.coe_smul, ennreal.smul_def, smul_eq_mul, ennreal.ess_sup_const_mul,
+    snorm_ess_sup]
 
```

Comment:
can you write a lemma  `snorm_comp_le` unifying your two statements? In this way, you could derive directly `snorm_const_smul_le` without separating the cases.

Reply:
Which two statements are you referring to? Does your comment still apply to the latest version?

Reply:
My comment was for a previous version. Now everything is great!